src/Contract/PheanstalkInterface.php

```diff
@@ -153,6 +153,13 @@ public function release(
      */
     public function reserve(): ?Job;
 
+    /**
+     * Reserves/locks a specific job, new in beanstalkd 1.12+
+     *
+     * @param JobIdInterface $job
+     */
+    public function reserveJob(JobIdInterface $job): Job;
```

Comment:
Just thinking out loud here...
If we add this to the interface, then any custom implementation of this interface will break.
So should we then, based on this, increase our major version number? Since this becomes a BC breaking release.

(PHPUnit explicitly marks what part of their classes are exempt from their BC promises; we could also do something like this in the future)

Reply:
Should probably try to solve this for the future as its potential that new commands could be added. Mark this interface as non-BC guaranteed and also bump a new major version?

Reply:
If that is the case I will also bump to 7.4.

Alternatively we can version the interface.

Like adding a new protocol1.12 interface that we implement and other implementers can then lag behind.

We could even just have a ReserveJobInterface or something like that... Just thinking out loud, don't go changing all your code unless you think it's the best ;)

Reply:
beandstalkd repo has a "wait", "put-with-id", "reserve-id" commands listed in issues but no PRs yet.